File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"
#include <string>

using namespace std;

namespace raindrops {
    string convert(int num){
        string result;
        bool divisible = false;
        
        if(num % 3 == 0){
            result += "Pling";
            divisible = true;
        } 
        
        if(num % 5 == 0){
            result += "Plang";
            divisible = true;
        }

        if(num % 7 == 0){
            result += "Plong";
            divisible = true;
        }

        if(!divisible){
            result = to_string(num);
        }

        return result;
    }
}  // namespace raindrops

File: solutions/python/raindrops/1/raindrops.py
def convert(number):
    # return a string
    # 3 -> "Plong"
    # 5 -> "Plang"
    # 7 -> "Plong"
    # none of them -> int stringyfied
    res = ""

    if number % 3 == 0:
        res += "Pling"

    if number % 5 == 0:
        res += "Plang"

    if number % 7 == 0:
        res += "Plong"

    if res == "":
        return str(number)
    
    return res
